### DISCUSSION TITLE
Does this work when the gui uses a unix socket?

### DISCUSSION THREAD

**Discussion by Kropatz** [Martchus/syncthingtray#249] [Q&A]
I setup my syncthing to provide the gui as a socket under /synced/gui-socket.
Tried setting that path in the tray application as the "Syncthing URL" and provided the API key, but that didn't work.
Is there a different way to specify the path or does the tray app not support interacting with the socket?

**Comment by Martchus** [comment]
Syncthing Tray uses Qt's HTTP library so I guess this won't work. Maybe there's a way to workaround it somehow (with minor code changes) but I wouldn't know out of my head.

**Reply by Martchus** [reply]
Considering https://bugreports.qt.io/browse/QTBUG-102855 this is not easily possible. Maybe it would be best to create a feature request for implementing this in Qt Network itself.

**Reply by Martchus** [reply]
Looks like there's actually some effort going on, e.g. see https://codereview.qt-project.org/c/qt/qtbase/+/550202 and the already referenced QTBUG. So I guess all we have to do it wait for this feature landing in Qt (and in case you're using distribution-provided packaging and update of those packages).

**Reply by Martchus** [reply]
Looks like this will make it into Qt 6.8.0 (https://codereview.qt-project.org/c/qt/qtbase/+/550202 has just been merged).

**Reply by Kropatz** [reply]
I have qt 6.8.0 now, but it doesn't seem to work unfortunately.
Tried http, unix+http and local+http.
Error is "Unable to request Syncthing config: host  not found"
![image](https://github.com/user-attachments/assets/c23b6610-eedd-469e-bafb-5c5e79309a57)

**Reply by Martchus** [reply]
Maybe I'll have to enable it from the application-side as well. However, the diff on Gerrit doesn't look that way. The only documentation I found are unfortunately just the [release notes](https://doc.qt.io/qt-6/whatsnew68.html).

The feature might still be buggy. Or it is a simple permission error.

**Reply by Kropatz** [reply]
Think it might be a bug, works fine when I use curl for example
```
$ curl --header "X-API-Key: $API_KEY" --unix-socket /synced/gui-socket http://localhost/rest/system/status
{
  "alloc": 34579152,
  "connectionServiceStatus": {
    "quic://0.0.0.0:22000": {
      "error": null,
      "lanAddresses": [
...
...
```

Added some logging and everything looks fine, but it doesn't work:
```
Info: Supported schemes: file, qrc, http, unix+http, local+http, https, data
Info: Querying API: GET unix+http:///synced/gui-socket/rest/system/status
Error: Syncthing API error: Host  not found
```

I'll wait for a new Qt version, or at least until there are some examples/docs :smile:

**Reply by Martchus** [reply]
Despite Qt's generally great documentation I wouldn't wait for a documentation change.

Maybe it is fixed in 6.8.1 which will be release soon. If not, it might be a good idea to come  up with a minimum reproducer and file a bug (or maybe there's already an open bug).

**Reply by Martchus** [reply]
Another user opened an issue for this: https://github.com/Martchus/syncthingtray/issues/338

It would probably make sense to continue further discussion there.
